main.go

```diff
@@ -73,6 +99,19 @@ func checkFiles(w io.Writer, rootRule rules.Rule, cfg *config.Config, files []so
 		// `allowUnmatched` is explicitly set to `true` in the configuration.
 		if !result.Success || (result.Unmatched.Size() > 0 && !cfg.AllowUnmatched) {
 			problematicFiles = append(problematicFiles, file.Path)
+
+			sarifResult := rules.FormatSarif(result)
+			sarifResult.File = file.Path
+			sarifResult.Message = result.Format()
```

Comment:
You could do that already inside `FormatSarif`.

Reply:
true, fixed